stanify/builders/stan_block_builder.py

```diff
@@ -0,0 +1,673 @@
+            self.code += "for (r in 1:R){\n"
+            self.code.indent_level += 1
+            for statement in self.stan_model_context.sample_statements:
+                if statement.lhs_expr in self.stan_model_context.obs_integ_outcome_vector_names:
+                    obs_vec_name = statement.lhs_expr
+                    target_vec_name = obs_vec_name[:-4]
+                    scale = statement.distribution_args[1]
+                    if statement.distribution_type in ["normal", "lognormal"]:
+                        self.code += f"loglik += {statement.distribution_type}_lpdf({obs_vec_name}[:, r]|{target_vec_name}[:, r], {scale});\n"
+                    elif statement.distribution_type in ["neg_binom_2"]:
+                        self.code += f"loglik += {statement.distribution_type}_lpmf({obs_vec_name}[:, r]|{target_vec_name}[:, r], {scale});\n"
+            self.code.indent_level -= 1
+            self.code += "}\n"
+
+    def build_post_pred_rng_functions(self):
+
+        if self.precision_context.R == 1:
+            self.code += "// Define and assign generated value to posterior predictive vector\n"
+            for statement in self.stan_model_context.sample_statements:
+                if statement.lhs_expr in self.stan_model_context.obs_integ_outcome_vector_names:
+                    # TODO @Dashadower how to use the following in the future?
+                    #  stan_type = self.stan_model_context.stan_data[statement.lhs_expr].stan_type #     stan_type = self.stan_model_context.stan_data[statement.lhs_expr].stan_type KeyError: 'prey_obs'
+                    scale = statement.distribution_args[1]
+                    # if stan_type.startswith("vector"):
+                    #     self.code += f"{stan_type} {statement.lhs_expr}_post = to_vector({statement.distribution_type}_rng({', '.join(statement.distribution_args)}));\n"
+                    # else:
+                    #     self.code += f"{stan_type} {statement.lhs_expr}_post = {statement.distribution_type}_rng({', '.join(statement.distribution_args)});\n"
+                    self.code += f"array[N] real {statement.lhs_expr}_post = {statement.distribution_type}_rng({', '.join(statement.distribution_args)});\n"
+        else:
+            self.code += "// Define observed vector (matching vector)\n"
+            for statement in self.stan_model_context.sample_statements:
+                if statement.lhs_expr in self.stan_model_context.obs_integ_outcome_vector_names:
+                    self.code += f"array[N] vector[R] {statement.lhs_expr}_post;\n"
+
+            self.code += "// Assign generated value to observed vector (matching vector)\n"
+            self.code += "for (r in 1:R){\n"
+            self.code.indent_level += 1
+            for statement in self.stan_model_context.sample_statements:
+                #TODO @Dashadower statement.lhs_variable vs .lhs_expr, obs_integ_outcome_vector_names is list VS target_// is tuple so the latter doesn't work
```

Comment:
> statement.lhs_variable vs .lhs_expr, obs_integ_outcome_vector_names is list VS target_// is tuple so the latter doesn't work

I'm not sure what you mean by this. Do you mean that you can't check for membership in a tuple? Because you can.